src/jdk.jpackage/windows/classes/jdk/jpackage/internal/WinMsiBundler.java

```diff
@@ -185,7 +185,18 @@ public class WinMsiBundler  extends AbstractBundler {
             new StandardBundlerParam<>(
                     "win.msi.productVersion",
                     String.class,
-                    VERSION::fetchFrom,
+                    params -> {
+                        String version = VERSION.fetchFrom(params);
+                        // Special case for MSI product version for runtime
+                        // installer. Runtime version can be single digit
+                        // for example "25", but product version requires 2 or 4
+                        // components. JDK uses "25.0.0.0" in this case.
+                        if (StandardBundlerParam.isRuntimeInstaller(params) &&
+                                !version.contains(".")) {
+                            version = version.concat(".0.0.0");
+                        }
+                        return version;
+                    },
```

Comment:
I guess, this is a workaround for the case when the version comes from JDK's release file. This is the wrong place for this workaround. It should be a part of the code reading version from JDK's release file.

Reply:
Yes, this is workaround when the version comes from JDK's release file. In this case name will be customeJDK-25.0.0.0 for example if we change it when we read version form file. Other platform do not need it. Only PRODUCT_VERSION in MSI needs to be 2 or 4 components, so I think it is correct location for it.

Reply:
For runtime packaging, the version can be taken from the command line or the "release" file. In the latter case the version should be valid for the current platform and packaging type. Version adjustments should be isolated in a function that reads a version from the "release" file. The result of this function call should be such that it will be used as-is without any modifications.

Ideally, we should have `Optional<String> readVersionFromRuntimeReleaseFile(String packageType, Path pathToReleaseFile)` function that we can unit test and that will return a valid version for the given packaging type ("packageType" parameter). It should return an empty Optional instance if it fails.